File: android/core/src/main/java/app/visly/vml/viewimpl/SolidColorViewImpl.kt
/**
 * Copyright (c) Visly Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package app.visly.vml.viewimpl

import android.content.Context
import android.view.View
import app.visly.vml.JsonValue
import app.visly.vml.Size

class SolidColorViewImpl(ctx: Context): BaseViewImpl<View>(ctx) {
    override fun measure(width: Float?, height: Float?): Size {
        return Size(width ?: 0f, height ?: 0f)
    }

    override fun setProp(key: String, value: JsonValue) {
        super.setProp(key, value)
    }

    override fun createView(): View {
        return View(ctx)
    }

    override fun bindView(view: View) {
        super.bindView(view)
    }
}

File: ios/ShardKit/Classes/SolidColorViewImpl.swift
/**
 * Copyright (c) Visly Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import UIKit

internal class SolidColorViewImpl: BaseViewImpl {
    override func measure(width: CGFloat?, height: CGFloat?) -> CGSize {
        return CGSize(width: width ?? 0, height: height ?? 0)
    }
    
    override func setProp(key: String, value: JsonValue) {
        super.setProp(key: key, value: value)
    }
    
    override func createView() -> UIView {
        return UIView()
    }
    
    override func bindView(_ view: UIView) {
        super.bindView(view)
    }
}
